Modules/_dbmmodule.c

```diff
@@ -78,14 +86,22 @@ dbm_dealloc(dbmobject *dp)
 {
     if ( dp->di_dbm )
         dbm_close(dp->di_dbm);
-    PyObject_Del(dp);
+    PyTypeObject *tp = Py_TYPE(dp);
+    freefunc free_func = PyType_GetSlot(tp, Py_tp_free);
+    free_func(dp);
+    Py_DECREF(tp);
```

Comment:
This change is unrelated to "Port _dbm module to multiphase initialization". I would prefer to see it in a separated PR. Also, currently `type->tp_free` is still preferred. There is no plan to move stdlib C extensions to the limited C API. If you have such plan, please open an issue to discuss this idea.

Reply:
@vstinner Hmm without this change this PR will cause the leak, but is it okay?

Reply:
You're right, ignore my comment.

I didn't notice that your change also converts a static type into a heap type. You should elaborate your commit messages: list all major changes. This PR is not just "Port _dbm module to multiphase initialization".